src/nvim/ex_getln.c

```diff
@@ -2436,7 +2436,7 @@ static void cmdpreview_restore_state(CpInfo *cpinfo)
     extmark_clear(buf, (uint32_t)cmdpreview_ns, 0, 0, MAXLNUM, MAXCOL);
   }
 
-  for (size_t i = 0; i < cpinfo->win_info.size; i++) {
+  for (size_t i = cpinfo->buf_info.size; i-- > 0;) {
```

Comment:
?

Reply:
My apologies for the copy-paste mistake; I've pushed a correction.  So much time spent on so many lines of explanation, only to fail on the tiny patch :-)

Reply:
And the explanation is outdated, because it has already been fixed by #24501 :)

Reply:
It looks like you have all of this covered and this pull request can be closed.  Thanks for your patience :-)